model/Image.php

```diff
@@ -328,6 +355,105 @@ public function generateSetSize(Image_Backend $backend, $width, $height) {
 		return $backend->paddedResize($width, $height);
 	}
 
+	/**
+	 * Proportionally scale down this image if it is wider or taller than the specified dimensions.
+	 * Similar to SetRatioSize but without up-sampling. Use in templates with $ScaleMax.
+	 * 
+	 * @uses Image::SetRatioSize()
+	 * @param integer $width The maximum width of the output image
+	 * @param integer $height The maximum height of the output image
+	 * @return Image
+	 */
+	public function ScaleMax($width, $height) {
+		return $this->width > $width || $this->height > $height
```

Comment:
Can you please use the `->getWidth()` and `->getHeight()` methods for the sake of code completion?

Also another discussion we had earlier in another issue, but we feel that getter methods are best invoked explicitly where available. :D